#! /bin/sh /usr/share/dpatch/dpatch-run
## 40_dryrunmore.dpatch by Petter Reinholdtsen <pere@hungry.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Make sure dryrun mode keep running even if one of the expected
## DP: rc directories (on debian typically /etc/rc.boot/) is missing.
## DP: Updated for version 1.02.0.

@DPATCH@
--- insserv-1.02.0.orig/insserv.c
+++ insserv-1.02.0/insserv.c
@@ -851,7 +851,7 @@
     }
 
     if ((rcdir = opendir(rcpath)) == NULL)
-	error("can not opendir(%s): %s\n", rcpath, strerror(errno));
+	warn("can not opendir(%s): %s\n", rcpath, strerror(errno));
 
     return rcdir;
 }
@@ -1069,6 +1069,8 @@
 	}
 
 	rcdir = openrcdir(rcd); /* Creates runlevel directory if necessary */
+	if (rcdir == NULL)
+	    break;
 	pushd(rcd);
 	while ((d = readdir(rcdir)) != NULL) {
 	    char * ptr = d->d_name;
@@ -2020,6 +2022,8 @@
 
 	script = NULL;
 	rcdir = openrcdir(rcd); /* Creates runlevel directory if necessary */
+	if (rcdir == NULL)
+	    break;
 	pushd(rcd);
 
 	/*
@@ -2181,6 +2185,8 @@
 
 	script = NULL;
 	rcdir = openrcdir(rcd); /* Creates runlevel directory if necessary */
+	if (rcdir == NULL)
+	    break;
 	pushd(rcd);
 
 	/*
